### PR TITLE
Update Kafka source/sink to use ByteCount

### LINKED ISSUE #3116: Update Kafka source/sink to use ByteCount
**Is your feature request related to a problem? Please describe.**

As noted [here](https://github.com/opensearch-project/data-prepper/pull/3115#discussion_r1284743547) the Kafka source/sink are using bytes without the Data Prepper byte type.

**Describe the solution you'd like**

Update all byte configurations to use the byte type - `ByteCount`.

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
Update Kafka source/sink to use ByteCount

Modified Kafka Topic Configuration to use ByteCount for bytes related configuration.
 Resolves #3116 
### Issues Resolved
Resolves #3116 
 
### Check List
- [X ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [X ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/TopicConfig.java [hunk 1]
```diff
@@ -73,22 +74,18 @@ public class TopicConfig {
     private Duration threadWaitingTime = DEFAULT_THREAD_WAITING_TIME;
 
     @JsonProperty("max_partition_fetch_bytes")
-    private Integer maxPartitionFetchBytes = DEFAULT_MAX_PARTITION_FETCH_BYTES;
+    private String maxPartitionFetchBytes = DEFAULT_MAX_PARTITION_FETCH_BYTES;
```
Comment (dlvenable) [inline data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/TopicConfig.java:77]: Data Prepper should be able to deserialize these natively. But, we can always clean this up later.

**Review by dlvenable — APPROVED** [review]

File: data-prepper-plugins/kafka-plugins/src/test/resources/sample-pipelines.yaml [hunk 2]
```diff
@@ -15,12 +15,12 @@ log-pipeline:
...
           heart_beat_interval: PT3S
-          fetch_max_bytes: 52428800
+          fetch_max_bytes: "50mb"
           fetch_max_wait: 500
-          fetch_min_bytes: 1
+          fetch_min_bytes: "1b"
           retry_backoff: PT100S
           consumer_max_poll_records: 500
-          max_partition_fetch_bytes: 10485760
+          max_partition_fetch_bytes: "10mb"
```
Comment (hshardeesi) [inline data-prepper-plugins/kafka-plugins/src/test/resources/sample-pipelines.yaml:23]: Default is 1mb

**Review by hshardeesi — APPROVED** [review]
max_partition_fetch_bytes in sample pipeline yaml should be 1 mb

File: data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/TopicConfig.java [hunk 3]
```diff
@@ -24,12 +25,12 @@ public class TopicConfig {
...
     static final Duration DEFAULT_MAX_RECORD_FETCH_TIME = Duration.ofSeconds(4);
-    static final Integer DEFAULT_FETCH_MAX_BYTES = 52428800;
+    static final String DEFAULT_FETCH_MAX_BYTES = "50mb";
     static final Integer DEFAULT_FETCH_MAX_WAIT = 500;
-    static final Integer DEFAULT_FETCH_MIN_BYTES = 1;
+    static final String DEFAULT_FETCH_MIN_BYTES = "1b";
     static final Duration DEFAULT_RETRY_BACKOFF = Duration.ofSeconds(10);
     static final Duration DEFAULT_RECONNECT_BACKOFF = Duration.ofSeconds(10);
-    static final Integer DEFAULT_MAX_PARTITION_FETCH_BYTES = 1048576;
+    static final String DEFAULT_MAX_PARTITION_FETCH_BYTES = "1mb";
```
Comment (hshardeesi) [inline data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/TopicConfig.java:33]: Should we make it configurable at byte level instead>? to be at par with kafka config.

**Pull request merged**